util/appdata.go

```diff
@@ -0,0 +1,101 @@
+package util
```

Comment:
Overall great job. please consider moving this to `filesystem` and check if the mechanism doesn't already exist there, if so maybe replace with this function.